tests/v2/validation/pipeline/Jenkinsfile.release.upgrade.local

```diff
@@ -146,19 +153,72 @@ node {
           dir ("./") {
             try {
               stage('Configure and Build') {
+                config = config.replace('${AWS_SECRET_ACCESS_KEY}', env.AWS_SECRET_ACCESS_KEY)
+                config = config.replace('${AWS_ACCESS_KEY_ID}', env.AWS_ACCESS_KEY_ID)
+                config = config.replace('${AWS_IAM_PROFILE}', env.AWS_IAM_PROFILE)
+                config = config.replace('${AWS_REGION}', env.AWS_REGION)
+                config = config.replace('${AWS_VPC}', env.AWS_VPC)
+                config = config.replace('${AWS_SECURITY_GROUPS}', env.AWS_SECURITY_GROUPS)
+                config = config.replace('${ADMIN_PASSWORD}', env.ADMIN_PASSWORD)
+            
                 if (env.AWS_SSH_PEM_KEY && env.AWS_SSH_KEY_NAME) {
                   dir("./rancher/tests/v2/validation/.ssh") {
                     def decoded = new String(AWS_SSH_PEM_KEY.decodeBase64())
                     writeFile file: AWS_SSH_KEY_NAME, text: decoded
                   }
                 }
-                dir("./rancher/tests/v2/validation") {
-                  def filename = "config.yaml"
-                  def configContents = env.CONFIG
+                
+                scriptPath = "./rancher/tests/v2/validation/pipeline/scripts/rancher_k8s_version.sh"
+                
+                def rancherK8sVersions = ""
+                try {
+                  rancherK8sVersions = sh(
+                    script: """
+                      if [ -f ${scriptPath} ]; then
+                      chmod +x ${scriptPath}
+                      ${scriptPath}
+                      else
+                        echo "Script not found!"
+                        exit 1
+                    fi
+                    """,
+                    returnStdout: true
+                  ).trim()
+
+                  def lines = []
+                  if (rancherK8sVersions != "") {
+                    lines = rancherK8sVersions.split('\n')
+                  }
 
-                  writeFile file: filename, text: configContents
-                  env.CATTLE_TEST_CONFIG = workPath+filename
+                  if (lines.size() >= 8) {
+                    def rancherImageTag = lines[0].replace("Latest Rancher Version: ", "").trim()
+                    def rancherVersion = rancherImageTag.replace("v", "")
+                    def preUpgradeRKE2K8s = lines[2]?.trim()
+                    def postUpgradeRKE2K8s = lines[3]?.trim()
+                    def preUpgradeK3sK8s = lines[4]?.trim()
+                    def postUpgradeK3sK8s = lines[5]?.trim()
+                    def preUpgradeRKE1K8s = lines[6]?.trim()
+                    def postUpgradeRKE1K8s = lines[7]?.trim()
```

Comment:
since you can control this earlier, why not do the trims in the rancher_k8s_version.sh script instead of here? It would be cleaner to do this in the script.

Reply:
Even if we add trimming in the shell script, we still need to trim here to make sure no additional spaces are adeed while the script is executed.